homework/index.js

```diff
@@ -30,15 +34,147 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+  function showRepoAndContributorsInDetail(obj, parents = {}) {
+    if (!parents.forRepo || !parents.forContributors) {
+      throw new Error(`Error: No parent is specified.`);
+    }
+
+    parents.forRepo.innerHTML = '';
+    parents.forContributors.innerHTML = '';
+    const table = createAndAppend('table', parents.forRepo);
+
+    Object.keys(obj).forEach(key => {
+      const value = obj[key];
+      if (value !== null) {
```

Comment:
You can return early here in case `value === null` (saves you the indentation and keeping track of where it should end) and may want to consider replacing the multiple `if` statements that test the same variable against a number of constants with a `switch` statement:

```js
Object.keys(obj).forEach(key => {
  const value = obj[key];
  if (value === null) {
    return;
  }
  let tableRow;
  switch (key) {
    case 'name':
      {
        tableRow = createAndAppend('tr', table, { class: 'tr' });
        createAndAppend('td', tableRow, {
          text: 'repository :',
          class: 'title',
        });
        const tdName = createAndAppend('td', tableRow);
        createAndAppend('a', tdName, {
          text: value,
          href: obj.repoUrl,
          target: '_blank',
        });
      }
      break;
    case 'description':
    case 'forks':
      tableRow = createAndAppend('tr', table, { class: 'tr' });
      createAndAppend('td', tableRow, {
        text: `${key} :`,
        class: 'title',
      });
      createAndAppend('td', tableRow, { text: value });
      break;
    case 'updated':
      tableRow = createAndAppend('tr', table, { class: 'tr' });
      createAndAppend('td', tableRow, {
        text: 'updated :',
        class: 'title',
      });
      createAndAppend('td', tableRow, {
        text: `${value.slice(0, 10)}, ${value.slice(11, 19)}`,
      });
      break;
    default:
  }
});
```